test/suites/chaos/zonal_failure_test.go

```diff
@@ -0,0 +1,489 @@
+						},
+					},
+				})
+				g.Expect(err).NotTo(HaveOccurred())
+
+				var instanceIDs []string
+				for _, reservation := range describeInstancesOutput.Reservations {
+					for _, instance := range reservation.Instances {
+						if instance.InstanceId != nil {
+							instanceIDs = append(instanceIDs, *instance.InstanceId)
+						}
+					}
+				}
+
+				if len(instanceIDs) > 0 {
+					fmt.Printf("Terminating %d remaining EC2 instances\n", len(instanceIDs))
+					_, err := env.EC2API.TerminateInstances(env.Context, &ec2.TerminateInstancesInput{
+						InstanceIds: instanceIDs,
+					})
+					g.Expect(awserrors.IgnoreNotFound(err)).NotTo(HaveOccurred())
+				}
+			}, 5*time.Minute).Should(Succeed())
+		},
+		Entry("complete failure", "100", "complete-failure-app", "complete AZ failure"),
+		Entry("moderate failure", "50", "moderate-failure-app", "moderate AZ failures"),
+		Entry("minor failure", "25", "minor-failure-app", "minor AZ failures"),
+	)
+})
+
+func createExperimentTemplate(ctx context.Context, env *awsenv.Environment, targetAZ string, instances []ec2types.Instance, failurePercentage string) string {
+	// Filter instances to only include those in the target AZ
+	var targetInstances []string
+	for _, instance := range instances {
+		if instance.InstanceId != nil && instance.Placement != nil && instance.Placement.AvailabilityZone != nil {
+			targetInstances = append(targetInstances, *instance.InstanceId)
+		}
+	}
+
+	subnetARNs := getSubnetsInAZ(ctx, env, targetAZ)
+	By(fmt.Sprintf("Found %d subnets in AZ %s", len(subnetARNs), targetAZ))
+
+	template := &fis.CreateExperimentTemplateInput{
+		Actions: map[string]fistypes.CreateExperimentTemplateActionInput{
+			"stop-instances": {
+				ActionId: aws.String("aws:ec2:stop-instances"),
+				Parameters: map[string]string{
+					"startInstancesAfterDuration": "PT5M", // Start instances after 5 minutes
+				},
+				Targets: map[string]string{
+					"Instances": "target-instances",
+				},
+			},
+			"ec2-capacity-error": {
+				ActionId: aws.String("aws:ec2:api-insufficient-instance-capacity-error"),
+				Parameters: map[string]string{
+					"availabilityZoneIdentifiers": targetAZ,
+					"duration":                    "PT5M",
+					"percentage":                  failurePercentage,
+				},
+				Targets: map[string]string{
+					"Roles": "target-roles",
+				},
+			},
+			"disrupt-subnet": {
+				ActionId: aws.String("aws:network:disrupt-connectivity"),
+				Parameters: map[string]string{
+					"duration": "PT5M",
+					"scope":    "all",
+				},
+				Targets: map[string]string{
+					"Subnets": "target-subnets",
+				},
+			},
+		},
+		Targets: map[string]fistypes.CreateExperimentTemplateTargetInput{
+			"target-instances": {
+				ResourceType:  aws.String("aws:ec2:instance"),
+				SelectionMode: aws.String("ALL"),
+				ResourceArns: lo.Map(targetInstances, func(id string, _ int) string {
+					return fmt.Sprintf("arn:aws:ec2:%s:%s:instance/%s", env.Region, env.ExpectAccountID(), id)
```

Comment:
Can use tags instead of ids? This way we can have one experiment for many instances. This will also make it less leak  the experiments